### ISSUE TITLE
Generated Data Shape always 0

### ISSUE THREAD

**Issue by iamamiramine** [zhao-zilong/Tabula#9]
I am facing an issue when generating data using Tabula. 

I trained Tabula on the following datasets:
1. **[Census](https://archive.ics.uci.edu/dataset/117/census+income+kdd)**
2. **[Fake Hotel Guests](https://www.kaggle.com/datasets/bhavkaur/hotel-guests-dataset)** 
3. **[Adult](https://archive.ics.uci.edu/dataset/2/adult)**
4. **[Health](https://www.kaggle.com/datasets/thedevastator/national-health-and-nutrition-examination-survey?resource=download&select=csv-1.csv)**
5. **[News](https://archive.ics.uci.edu/dataset/332/online+news+popularity)**

However, when generating, the generation loop is stuck because generated data shape is always 0 (`num_samples` is always greater than `gen_data.shape[0]`).

I tried re-training, and tried changing the `max_length` parameter in the sampling function, but it was of no help.

Can you please help me figure out how to fix this issue?

**Comment by zhao-zilong** [comment]
Hi @iamamiramine  sorry that I just saw your message. Did you solve it? The reason can be that your max_length is too small so that the generation cannot successfully generate one complete row of data.

**Comment by iamamiramine** [comment]
Hello, I tried changing the max_length parameter and it did not work.
Another thing to note is that Fake Hotel Guests dataset consists of 9 columns, so one row from this dataset is relatively short.

**Comment by omaralvarez** [comment]
I am also having problems with this, I am using `max_length=1024` the maximum, if use more I get a CUDA error, in this dataset I can not get a single sample:

``` python
from imblearn.datasets import fetch_datasets

sick = fetch_datasets()['sick']
sick.data.shape
```

**Comment by zhao-zilong** [comment]
Hi @omaralvarez  @iamamiramine  

You do not need to set the max_length to 1024 that big, you can uncomment this part of code to see what is the length of your encoded row:

https://github.com/zhao-zilong/Tabula/blob/3869567c681b6c7cb1051bed75b5cb9ccfd2fa3a/tabula/tabula_dataset.py#L64

Let me know if that helps.

**Comment by omaralvarez** [comment]
Yes, I don't think it has to do with max_length, the issue in this case is that some numbers always are outside of the requested ranges in the predicted dataframe, so they are always filtered out. I have tried to switch temperature, k, and training epochs to no avail.

**Comment by xxxx-lzw** [comment]
> I am facing an issue when generating data using Tabula.
> 
> I trained Tabula on the following datasets:
> 
> 1. **[Census](https://archive.ics.uci.edu/dataset/117/census+income+kdd)**
> 2. **[Fake Hotel Guests](https://www.kaggle.com/datasets/bhavkaur/hotel-guests-dataset)**
> 3. **[Adult](https://archive.ics.uci.edu/dataset/2/adult)**
> 4. **[Health](https://www.kaggle.com/datasets/thedevastator/national-health-and-nutrition-examination-survey?resource=download&select=csv-1.csv)**
> 5. **[News](https://archive.ics.uci.edu/dataset/332/online+news+popularity)**
> 
> However, when generating, the generation loop is stuck because generated data shape is always 0 ( is always greater than ).`num_samples``gen_data.shape[0]`
> 
> I tried re-training, and tried changing the parameter in the sampling function, but it was of no help.`max_length`
> 
> Can you please help me figure out how to fix this issue?

May I ask if your problem is solved, I'm experiencing this problem as well

**Comment by tmacleod** [comment]
This problem also occurs if you train with too few epochs. Train with more epochs and sampling speed improves.

**Comment by omaralvarez** [comment]
In my case, no matter the epochs it wouldn't work (I trained for a week in a 80GB A100).

**Comment by zhao-zilong** [comment]
@omaralvarez @xxxx-lzw @iamamiramine @tmacleod  I currently solved a case that also enters the infinite generation loop. It is not a grace solution, but it kind of solved the problem. 

First in this line:
https://github.com/zhao-zilong/Tabula/blob/1c80a566e26fd28187d25929436d01ddf5dc751d/tabula_middle_padding/tabula_dataset.py#L101

add `print(batch["labels"].shape)` and then run the training code.

For example, if you see `[16, 610]` as the batch shape, then 610 is the sequence length of the encoded row; use it as the `max_length` in your generation, which can eventually generate data.
